tests/deployment_test.go

```diff
@@ -291,53 +270,97 @@ func (d *DeploymentSuite) validateListDeploymentsResponse_WithSeriesFilter(ctx c
 			return
 		}
 		a.NotNil(resp.Deployments)
-		a.Equal(expectedNumberOfDeployments, len(resp.Deployments))
 
 		// check to stop eventuallyWithT from panicking since
 		// it collects all possible errors
 		if len(resp.Deployments) < 1 {
 			return
 		}
 
-		deployment := resp.Deployments[0]
-		a.Equal(expectedDeployment.SeriesName, deployment.Deployment.SeriesName)
-		a.Equal(expectedDeployment.BuildId, deployment.Deployment.BuildId)
-		a.Equal(false, deployment.IsCurrent)
+		for _, expectedDeploymentListInfo := range expectedDeployments {
+			if seriesFilter != "" && expectedDeploymentListInfo.Deployment.SeriesName != seriesFilter {
+				// don't need to validate those DeploymentListInfo with seriesName != seriesFilter
+				continue
```

Comment:
This should fail the test, if there is a series filter, there should not be any result from other series.

Reply:
yes I agree - however, this modification (of skipping over results) is being conducted on the structure I have made for verifying deployments. I am not altering the response structure - we basically want to verify if all the deployments we expect are what's being returned

Reply:
if you are excluding something from "expectedDeployments" then it should not be called "expected".